shell/server/pre-meteor.js

```diff
@@ -43,6 +43,12 @@ function isSandstormShell(hostname) {
   return (hostname === HOSTNAME || (DDP_HOSTNAME && hostname === DDP_HOSTNAME));
 }
 
+function hasPseudoTLD(hostname) {
+  // Returns true if this hostname ends in one of the major pseudo-TLDs.
+  let tld = hostname.split(".").pop();
+  return (tld === ("onion" || "i2p" || "gnu"));
```

Comment:
@paulproteus Yeah, sorry, you're totally right about the short-circuit evaluation on the `||` operator… ffffuuuuuu, I hate Javascript. I've added a patch to this branch which changes this function to use a mix between your suggested way and the original patch:
```
var tld = hostname.split(".").pop();
var pseudoTLDs = ["onion", "i2p", "gnu"];
return (_.contains(pseudoTLDs, tld));
```